methylprep/processing/read_geo_processed.py

```diff
@@ -737,3 +722,43 @@ def detect_header_pattern(test, filename, return_sample_column_names=False):
             'sample_names': sample_columns,
             'sample_names_stems': sample_column_residuals, # unique parts of sample names
            }
+
+def pd_load(filepath, **kwargs):
+    """ helper function that reliably loads any GEO file, by testing for the separator that gives the most columns """
+    this = Path(filepath)
+
+    if this.suffix not in ('.xlsx', '.pkl'):
+        # first, check that we're getting the max cols
+        test_csv = pd.read_csv(this, nrows=100, skiprows=kwargs.get('skiprows',0))
+        test_t = pd.read_csv(this, sep='\t', nrows=100, skiprows=kwargs.get('skiprows',0))
+        test_space = pd.read_csv(this, sep=r',\s+', quoting=csv.QUOTE_ALL, engine='python', skiprows=kwargs.get('skiprows',0))
```

Comment:
Shouldn't nrows=100 be added here as well? Without that the processing time may be doubled, as pd.read_csv is very slow for big files.